ndcube/utils/wcs.py

```diff
@@ -477,3 +478,28 @@ def compare_wcs_physical_types(source_wcs, target_wcs):
     target_wcs = get_low_level_wcs(target_wcs, 'target_wcs')
 
     return source_wcs.world_axis_physical_types == target_wcs.world_axis_physical_types
+
+
+def is_wcs_2d_celestial(wcs):
+    '''
+    Checks to see if the WCS object is 2D and has celestial axes only.
+
+    Parameters
+    ----------
+    wcs: `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
+        The input WCS object.
+
+    Returns
+    -------
+    result: `bool`
+    '''
+    wcs = get_low_level_wcs(wcs)
+
+    if wcs.world_n_dim > 2:
+        return False
+
+    for world_axis_class in wcs.world_axis_object_classes.values():
+        if not issubclass(world_axis_class[0], SkyCoord):
+            return False
+
+    return True
```

Comment:
But wouldn't we want all axes to be celestial in this case?
Otherwise I think we could make use of https://github.com/astropy/reproject/blob/master/reproject/wcs_utils.py#L203 directly.

Reply:
I think we need clarification from @Cadair here.  It seems if we want @Cadair's logic than using `has_celestial` is the was to go.  But you're right that if we are restricting this to 2-D celestial cubes only, then the above logic looks correct.

Reply:
Updated the code to use `has_celestial`.